### PR TITLE
feat: ANA-1445 Add additional configuration for setting up codeceptjs framework

### PR DESCRIPTION
**Pull request by avattipalli**
This change consists of additional setup required which includes the following:

- containerization of the application
- add circleci configuration
- Usage of yarn 
- Using husky and commitlint
- configuring eslint. Haven't added any rules to it yet.
- add .gitignore
- updates to package.json

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  full:
+    jobs:
+      - run_tests:
+          steps:
+            - install-dependencies
+            - run: yarn test
+      - build-assets:
+          requires:
+            - install-dependencies
+      - ci/build-container:
```
Comment (aaron-humerickhouse) [inline .circleci/config.yml:56]: This doesn't need to be containerized

File: .container.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+name: bc-functional-ui
```
Comment (aaron-humerickhouse) [inline .container.yml:1]: doesn't need to be containerized

File: gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+const gulp = require('gulp');
+const NgBuild = require('ng-build');
```
Comment (aaron-humerickhouse) [inline gulpfile.js:2]: I would avoid ngBuild, they're trying to get rid of it

File: package.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    "url": "git@github.com:bigcommerce-labs/bc-functional-ui"
+  },
+  "dependencies": {
+    "@codeceptjs/configure": "latest",
+    "codeceptjs": "^2.6.5"
+  },
+  "devDependencies": {
+    "@commitlint/cli": "^8.3.5",
+    "@commitlint/config-conventional": "^8.3.4",
+    "@wdio/cli": "^6.1.12",
```
Comment (aaron-humerickhouse) [inline package.json:28]: wdio is a dependency, not a dev dependency, as is webdriverio

File: package.json [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    "type": "git",
+    "url": "git@github.com:bigcommerce-labs/bc-functional-ui"
+  },
+  "dependencies": {
+    "@codeceptjs/configure": "latest",
+    "codeceptjs": "^2.6.5"
+  },
+  "devDependencies": {
+    "@commitlint/cli": "^8.3.5",
+    "@commitlint/config-conventional": "^8.3.4",
```
Comment (aaron-humerickhouse) [inline package.json:27]: `@bigcommerce/validate-commits` https://github.com/bigcommerce/validate-commits

File: package.json (hunk 4 above)
Comment (avattipalli) [reply package.json:28]: That kind of makes sense for this repo. I usually see testing frameworks being added as a dev dependency

File: gulpfile.js (hunk 3 above)
Comment (avattipalli) [reply gulpfile.js:2]: I didn't know about this, is there any recommendation to an alternate approach?

**Comment by avattipalli** [comment]
> A couple package things
> This does not need to be containerized

Made the necessary changes as suggested 👍

File: gulpfile.js (hunk 3 above)
Comment (aaron-humerickhouse) [reply gulpfile.js:2]: It depends on what you're using it for

Comment (avattipalli) [reply gulpfile.js]: As discussed offline, we don't think it is necessary to have ng-build. Removed gulp and its usage for now. Will add them if required at a later time.

File: .lobby.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+default: &default
```
Comment (aaron-humerickhouse) [inline .lobby.yml:1]: what's this file for?

File: config/build.sh [hunk 7]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
```
Comment (aaron-humerickhouse) [inline config/build.sh:1]: I don't think we need a build script

Comment (aaron-humerickhouse) [reply config/build.sh:1]: pretty sure it's for containerization and we have no public folder

File: .gitignore [hunk 8]
```diff
@@ -0,0 +1,13 @@
+# dependencies
+/node_modules
+
+# production
+/build
```
Comment (aaron-humerickhouse) [inline .gitignore:5]: there shouldn't be a build directory

File: .circleci/config.yml [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+            - install-dependencies
+            - run: yarn test
+      - build-assets:
+          requires:
+            - install-dependencies
+          pre-steps:
+            - ci/pre-setup
+            - attach_workspace:
+                at: *workspace_root
+      - ci/notify-success:
```
Comment (aaron-humerickhouse) [inline .circleci/config.yml:60]: this may fail if there's not a launchbay project.  I'm not sure

File: .circleci/config.yml [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+  install-dependencies:
+    <<: *node_executor
+    steps:
+      - install-dependencies
+
+  unit-tests:
+    <<: *node_executor
+    steps:
+      - install-dependencies
+      - run: yarn run ci:test
```
Comment (aaron-humerickhouse) [inline .circleci/config.yml:32]: this needs to be changed to `yarn run test`

File: .circleci/config.yml [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+
+  build-assets:
```
Comment (aaron-humerickhouse) [inline .circleci/config.yml:34]: this is unnecessary

File: .circleci/config.yml [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+      - persist_to_workspace:
+          root: *workspace_root
+          paths:
+            - dist
+
+workflows:
+  version: 2
+  full:
+    jobs:
+      - run_tests:
```
Comment (aaron-humerickhouse) [inline .circleci/config.yml:49]: change in favor of defined job unit-tests

File: .circleci/config.yml [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+          steps:
+            - install-dependencies
+            - run: yarn test
+      - build-assets:
```
Comment (aaron-humerickhouse) [inline .circleci/config.yml:53]: remove

File: .circleci/config.yml [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+          requires:
+            - install-dependencies
+          pre-steps:
+            - ci/pre-setup
+            - attach_workspace:
+                at: *workspace_root
+      - ci/notify-success:
+          requires:
+            - run_tests
+            - ci/build-container
```
Comment (aaron-humerickhouse) [inline .circleci/config.yml:63]: remove

**Comment by avattipalli** [comment]
> Still some cleanup to do

Thanks for the feedback 👍 will work on fixing it

File: .circleci/config.yml [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+
+workflows:
+  version: 2
+  full:
+    jobs:
+      - run_tests:
+          steps:
+            - unit-tests
+
+notify:
```
Comment (aaron-humerickhouse) [inline .circleci/config.yml:39]: if notify_success or notify_failure jobs aren't present, this isn't needed.

**Pull request merged**